gui/src/pages/WorkspaceNetworking.svelte

```diff
@@ -30,6 +30,16 @@
           getValue: (network) => network.name,
         },
       ]}
+      actions={[
+        {
+          tooltip: 'Delete network',
+          glyph: 'Delete',
+          callback: async (component) => {
+            await workspace.deleteComponent(component.id);
+            window.location.reload();
```

Comment:
Surely we can do better than refreshing the entire page when the data changes...

A few concrete suggestions here:

1) Try to give callbacks a more descriptive name. In many cases, that would be something like `onEventName`. Here, it might be something like `execute`.
2) Event callbacks should _usually_, accept an extensible `event` argument. If you call this `execute`, then you might want to just have it be `context`. This will allow you to have a place you can stash extra information if you need it. See next point.
3) You can pass control objects/functions as context to the callback. So you might have something like: `async execute({ reload }, component) { ... }` which allows the ComponentTable component to expose a non-full-page-refresh.

Reply:
Addressed (1) in #394